homeassistant/components/sensor/uber.py

```diff
@@ -0,0 +1,247 @@
+REQUIREMENTS = ['https://github.com/denismakogon/rides-python-sdk/archive/'
+                'py3-support.zip#'
+                'uber_rides==0.1.2-dev']
+
+ICON = 'mdi:taxi'
+
+# Return cached results if last scan was less then this time ago.
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=30)
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Uber sensor."""
+    if None in (config.get('start_latitude'), config.get('start_longitude')):
+        _LOGGER.error(
+            "You must set start latitude and longitude to use the Uber sensor!"
+        )
+        return False
+
+    if config.get('server_token') is None:
+        _LOGGER.error("You must set a server_token to use the Uber sensor!")
+        return False
+
+    from uber_rides.session import Session
+
+    session = Session(server_token=config.get('server_token'))
+
+    wanted_product_ids = config.get('product_ids')
+
+    dev = []
+    start_lat_long = LatLong(config.get('start_latitude'),
+                             config.get('start_longitude'))
+    end_lat_long = LatLong(config.get('end_latitude'),
+                           config.get('end_longitude'))
+    timeandpriceest = UberEstimate(session, start_lat_long, end_lat_long)
+    for product_id, product in timeandpriceest.products.items():
+        if wanted_product_ids is not None and product_id in wanted_product_ids:
+            dev.append(UberSensor('time',
+                                  timeandpriceest,
+                                  product_id, product))
+            dev.append(UberSensor('price',
+                                  timeandpriceest,
+                                  product_id, product))
+        elif wanted_product_ids is None:
+            dev.append(UberSensor('time',
+                                  timeandpriceest,
+                                  product_id,
+                                  product))
+            dev.append(UberSensor('price',
+                                  timeandpriceest,
+                                  product_id,
+                                  product))
+    add_devices(dev)
+
+
+# pylint: disable=too-few-public-methods
+class UberSensor(Entity):
+    """Implementation of an Uber sensor."""
+
+    def __init__(self, sensorType, products, product_id, product):
+        """Initialize the Uber sensor."""
+        self.data = products
+        self._product_id = product_id
+        self._product = product
+        self._sensortype = sensorType
+        self._name = self._product.get('display_name') + " " + self._sensortype
+        if self._sensortype == "time":
+            self._unit_of_measurement = "min"
+            self._state = int(self._product.get(
+                'time_estimate_seconds', 0
+            ) / 60)
+        elif self._sensortype == "price":
+            if self._product.get('price_details').get('low_estimate') is None:
+                self._unit_of_measurement = self._product.get(
```

Comment:
These kind of enters inside your `get(…)` statements make the code very unreadable. please get rid of all of them.